### ISSUE TITLE
supabase serve uses "--no-npm" mode for Deno. Why? This breaks code that needs npm modules.

### ISSUE THREAD

**Issue by evelant** [supabase/cli#942]
# Bug report

I ported my firebase functions to supabase edge functions. In doing so I had to use some npm modules, which I thought was fine since deno supports npm. Upon trying to run my code however I was suprised to be hit with 

```
worker thread panicked Unsupported scheme "npm" for module "npm:/@effect/io@0.7.0/Exit". Supported schemes: [
    "data",
    "blob",
    "file",
    "http",
    "https",
]
```

## Describe the bug

The supabase cli passes "--no-npm" when invoking Deno. 
Code is here https://github.com/supabase/cli/blob/a30210717bdb1f3c9d3d5ce940d000971fc1a6da/internal/functions/serve/serve.go#L190

As far as I can tell this isn't documented anywhere by Supabase resulting in a surprise for users. Since Supabase targets people who want to switch from Firebase this is likely to affect almost everyone who attempts to switch as most people use npm modules in their firebase functions.

## To Reproduce

1. Add an npm specifier to your import_map.json such as `"@effect/io/": "npm:/@effect/io@0.7.0/",`
2. run `supabase functions serve`
3. Observe that function crashes at runtime due to `--no-npm` option passed to Deno by supabase cli.

## Expected behavior

All Deno features should work. If there's a specific reason why `--no-npm` is needed it should be clearly documented along with an explanation of how to work around being unable to use npm import specifiers.

## Screenshots

N/A

## System information

- OS: macOS
- Browser N/A
- Version of supabase-js: N/A
- Version of Node.js: N/A

**Comment by evelant** [comment]
For additional context -- I need to use npm resolutions in Deno because there are some modules (like `@effect/io`) that tools like `esm.sh` choke on. The only way to load them in deno is via npm.

**Comment by sweatybridge** [comment]
Thank you for clarifying your use case. I've pinged our functions team to help take a look.

**Comment by laktek** [comment]
@evelant [Deno Deploy](https://deno.com/deploy), the hosting platform we use for functions doesn't support npm specifiers yet. To keep dev/prod parity we disable the npm resolution when serving functions locally.

We will update the docs to mention this limitation.

**Comment by evelant** [comment]
@laktek OK thanks. I'm running into another issue as well -- globals that should be defined are undefined in the edge runtime. Specifically `performance`. A library tries to access `performance.now()` and it crahes, but _only_ on the edge runtime. Run it locally with `deno run -A my_file.ts` and `performance` is accessible and works fine.

**Comment by evelant** [comment]
@laktek according to the deno deploy docs `performance` is an API available on the Deno deploy runtime. https://deno.com/deploy/docs/runtime-api so it appears to be a bug that it's missing in the Supabase edge-runtime.

I want to investigate further but I don't see the Dockerfile for the edge runtime available anywhere. Is it not released on github?

**Comment by evelant** [comment]
@sweatybridge @laktek I've looked everywhere and I haven't been able to find source for `edge-runtime`, could you point me to it?  I'd like to investigate the missing globals problem but without the source it's a black box.

**Comment by laktek** [comment]
@evelant we will be open-sourcing `edge-runtime` in the coming weeks.

> according to the deno deploy docs performance is an API available on the Deno deploy runtime. https://deno.com/deploy/docs/runtime-api so it appears to be a bug that it's missing in the Supabase edge-runtime.

`performance` API will be included in an upcoming release. We'll update this ticket when the release is made.

**Comment by evelant** [comment]
@laktek Can you provide any more information about what is/isn't available in the edge-runtime? The docs are pretty sparse at the moment.

**Comment by laktek** [comment]
> Deno.errors.BadResource is undefined. Is Deno.errors missing from the supabase edge runtime?

Yep, looks like `Deno.errors` are missing from the global scope. We'll make a new release to fix it.

BTW, you can check what's available in edge-runtime's global scope by adding `console.log(Object.keys(globalThis))` and more specifically for Deno namespace `console.log(Object.keys(Deno))` to your function code.
